### PR TITLE
Scroll/zoom changes for desktops (based on Julien's Android changes)

### PR DESCRIPTION
**Pull request by ammon-c**
This libmatoya branch will contain some changes to add DESKTOP scroll/zoom support, which are being adapted from the ANDROID scroll/zoom support that Julien worked on previously (e.g. from the Parsec "android-zoom" project which refers to several matoya branches - See Julien's PRs for details).

Note that this branch has a sister branch in the Parsec repo (same name "unscaled-view") that has the corresponding app-side changes that use the matoya changes.

Implementation: On Android the scroll/zoom was implemented with finger gestures, but on desktop the gestures don't apply so I need to add support for scrollbars or edge scrolling, and sending the corresponding zoom and scroll events back to the app.

### CODE REVIEW

**Comment by ammon-c** [comment]
Julien's MTY_Zoom and MTY_Cursor functions which were originally part of this PR are being moved out of libmatoya over to the app side.  So the bulk of the changes remaining in this PR are from Julien's changes to the quad rendering code to support scaling and offsetting of the quad image, plus a few additional changes to support the app's desktop zoom and scroll code.

File: src/unix/linux/x11/app.c [hunk 1]
```diff
@@ -1382,6 +1382,29 @@ bool MTY_WindowExists(MTY_App *app, MTY_Window window)
...
+
+	struct window *ctx = app_get_window(app, winid);
+	if (!ctx)
+		return false;
+
+	Window root_window = 0, child_window = 0;
+	int root_x = 0, root_y = 0, win_x = 0, win_y = 0;
+	unsigned mask = 0;
+	XQueryPointer(app->display, ctx->window, &root_window, &child_window,
+		&root_x, &root_y, &win_x, &win_y, &mask);
```
Comment (Namaneo) [inline src/unix/linux/x11/app.c:1398]:
```suggestion
	XQueryPointer(app->display, ctx->window, &root_window, &child_window, &root_x, &root_y, &win_x, &win_y, &mask);
```

File: src/windows/appw.c [hunk 2]
```diff
@@ -193,10 +193,10 @@ static void app_register_raw_input(USHORT usage_page, USHORT usage, DWORD flags,
...
 {
 	if (app->AdjustWindowRectExForDpi) {
-		app->AdjustWindowRectExForDpi(r, WS_OVERLAPPEDWINDOW, FALSE, 0, lrint(scale * 96));
+		app->AdjustWindowRectExForDpi(r, WS_CAPTION | WS_BORDER, FALSE, 0, lrint(scale * 96));
 
 	} else {
-		AdjustWindowRect(r, WS_OVERLAPPEDWINDOW, FALSE);
+		AdjustWindowRect(r, WS_CAPTION | WS_BORDER, FALSE);
 	}
 }
```
Comment (Namaneo) [inline src/windows/appw.c:194]: Is that something retrieve from another PR?

File: src/windows/appw.c [hunk 3]
```diff
@@ -2207,6 +2201,26 @@ bool MTY_WindowExists(MTY_App *app, MTY_Window window)
...
+		return false;
+
+	RECT window_rect;
+	if (!GetWindowRect(ctx->hwnd, &window_rect))
+		return false;
+
+	return (cursor_pos.x >= window_rect.left  &&
+			cursor_pos.x <= window_rect.right &&
+			cursor_pos.y >= window_rect.top   &&
+			cursor_pos.y <= window_rect.bottom);
```
Comment (Namaneo) [inline src/windows/appw.c:2184]:
```suggestion
	return (cursor_pos.x >= window_rect.left  &&
		cursor_pos.x <= window_rect.right &&
		cursor_pos.y >= window_rect.top   &&
		cursor_pos.y <= window_rect.bottom);
```

File: src/unix/web/app.c [hunk 4]
```diff
@@ -572,6 +572,11 @@ bool MTY_WindowExists(MTY_App *app, MTY_Window window)
 	return true;
 }
 
+bool MTY_WindowContainsCursor(MTY_App *app, MTY_Window window)
+{
+	return false;
```
Comment (Namaneo) [inline src/unix/web/app.c:577]: Shouldn't this one return `true` like Android?

**Review by Namaneo — CHANGES_REQUESTED** [review]

**Comment by ammon-c** [comment]
Closing this because the corresponding Parsec PR was closed ("unscaled-view" branch in the Parsec repo).  Will reopen this one later if/when that branch is reopened.

**Pull request closed without merging**